### PR TITLE
Paul/check mnemonic

### PR DESCRIPTION
**Pull request by paullinator**
### CHANGELOG

 none

### Dependencies

 none

### Description

 none

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204749597603194

### CODE REVIEW

File: src/bin/searchLogs.ts [hunk 1]
```diff
@@ -1,14 +1,33 @@
...
+// const w = checkme.split(' ')
+// let av = true
+// for (const word of w) {
+//   if (!wordlists.english.includes(word)) {
+//     av = false
+//     break
+//   }
+// }
+
+// console.log(`checkme = ${av.toString()}`)
```
Comment (samholmes) [inline src/bin/searchLogs.ts]: This commented out code is distracting and not necessary?

File: src/indexApi.ts [hunk 2]
```diff
@@ -374,6 +374,30 @@ function checkForKeys(data: any): void {
...
+      if (allValid) {
+        slackPoster(
+          `Log attempt rejected due to mnemonic seed detected ${words
+            .slice(0, 3)
+            .join('-')}`
+        ).catch(e => console.log(e.message))
+        throw new Error('Log includes sensitive mnemonic data')
+      }
+    }
+  }
```
Comment (samholmes) [inline src/indexApi.ts]: This seems fair to be de-duped into a util?

File: src/indexApi.ts [hunk 3]
```diff
@@ -374,6 +374,30 @@ function checkForKeys(data: any): void {
...
+
+  const regex = /(\b[a-z]+\b\s){11,23}\b[a-z]+\b/g
+  const matches = dataString.match(regex)
+  const found = matches != null
+  if (found) {
+    for (const match of matches) {
+      const words = match.split(' ')
+      let allValid = true
+      for (const word of words) {
+        if (!wordlists.english.includes(word)) {
```
Comment (samholmes) [inline src/indexApi.ts]: Since there are 2048 words, should we turn this into a map for faster lookup time?

**Review by samholmes — CHANGES_REQUESTED** [review]

File: src/bin/searchLogs.ts (hunk 1 above)
Comment (paullinator) [reply src/bin/searchLogs.ts]: done. this was a utility file for misc use but i cleaned this up anyway

File: src/indexApi.ts (hunk 2 above)
Comment (paullinator) [reply src/indexApi.ts]: done w/ new commit

File: src/indexApi.ts (hunk 3 above)
Comment (paullinator) [reply src/indexApi.ts]: done

**Review by samholmes — APPROVED** [review]

**Pull request merged**